packages/e2e/lib/e2e.dart

```diff
@@ -128,9 +137,57 @@ class E2EWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding {
   /// The default value is `null`.
   Map<String, dynamic> reportData;
 
+  /// Tests will put the action requests from WebDriver to this pipe.
+  Completer<WebDriverAction> webDriverActionPipe = Completer<WebDriverAction>();
+
+  /// Updated when WebDriver completes the request by the test method.
+  ///
+  /// For example, a test method will ask for a screenshot by calling
+  /// `takeScreenshot`. When this screenshot is taken [driverActionComplete]
+  /// will complete.
+  Completer<bool> driverActionComplete = Completer<bool>();
+
+  final Mutex webDriverCommandLock = new Mutex();
+
+  /// Takes screenshot using WebDriver screenshot command.
+  ///
+  /// Only works on Web when tests are run via `flutter driver` command.
+  ///
+  /// See: https://www.w3.org/TR/webdriver/#screen-capture
+  Future<void> takeScreenshot(String screenshot_name) async {
+    if (!kIsWeb) {
+      throw StateError('Web driver commands are only avaliable on web');
+    }
+    await _webDriverCommand(WebDriverAction.screenshot(screenshot_name));
+  }
+
+  Future<void> _webDriverCommand(WebDriverAction command) async {
```

Comment:
Shouldn't this go in a web-specific implementation file, and we'd have another implementation file for IO, so we just call the method (or instantiate the class) and the VM version just throws for now?

Reply:
I carried the webdriver specific logic to another file. Please have another look.